### DISCUSSION TITLE
Building a plugin to send messages to Telegram

### DISCUSSION THREAD

**Discussion by AlexeyZelenin** [datalust/seq-tickets#1333] [Help]
Hello,

we tried to use this plugin:
Seq.App.Telegram
https://github.com/stdray/Seq.App.Telegram
to send messages to telegram,
but we can’t access system fields (like "@t" or "@mt") and complex object fields like "context.MyField"
Looks like we have to adjust this regex:
https://github.com/stdray/Seq.App.Telegram/blob/master/src/Seq.App.Telegram/MessageFormatter.cs#L13

So we made a fork:
https://github.com/swisschain/Seq.App.Telegram
and a GitHub action to build it and publish to NuGet:
https://github.com/swisschain/Seq.App.Telegram/blob/master/.github/workflows/DeployNuget.yml
here it is:
https://www.nuget.org/packages/Swisschain.Seq.App.Telegram

but when we try to install our new plugin to Seq we have this error:
Could not install app
Unexpected character encountered while parsing value: <. Path “, line 0, position 0.

https://ibb.co/zFY8ptD

Seq 5.1.3000

Questions:
1. Is there a better plugin to send messages to telegram?
2. Why do we have the error while trying to install our plugin to Seq? Probably we are building it wrong, could you please suggest how to do it the right way with GitHub actions?

Thank you for your time,
we really love Seq!

**Comment by nblumhardt** [comment]
Hi Alexey, thanks for getting in touch.

The installation error is most likely due to the HTTP request out to the NuGet feed receiving HTML (or XML) when JSON is expected. One possibility might be a proxy between the Seq server and NuGet, redirecting unauthenticated requests to an error page. Worth checking out, if you're using a proxy.

There have been a few changes around feed handling (and TlS version support) in the Seq versions since 5.1. Are you able to update to 2021.2 and try the installation again?

Another option would be to create a local folder on your Seq server (or mount one, if it's in Docker), and then create a new NuGet feed using that folder as the feed location. You can then drop `.nupkg` files directly into the folder, and install them from the feed.

I'm not aware of a better Telegram plugin; it does look like the current _Seq.App.Telegram_ plug-in accepts PRs, though, so there may be some chance to have your changes incorporated there? Please feel free to `@` me on any discussion that would benefit from another set of eyes :-)

Best regards,
Nick
